projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -0,0 +1,185 @@
+  // Set up hi/lo-thresh registers. This particular setup enables the ALRT/RDY pin.
+  status_ok_or_return(prv_setup_register(storage, ADS1015_ADDRESS_POINTER_LO_THRESH,
+                                         ADS1015_LO_THRESH_REGISTER_MSB,
+                                         ADS1015_LO_THRESH_REGISTER_LSB));
+  status_ok_or_return(prv_setup_register(storage, ADS1015_ADDRESS_POINTER_HI_THRESH,
+                                         ADS1015_HI_THRESH_REGISTER_MSB,
+                                         ADS1015_HI_THRESH_REGISTER_LSB));
+  GPIOSettings gpio_settings = {
+    .direction = GPIO_DIR_IN,  //
+  };
+  InterruptSettings it_settings = {
+    .type = INTERRUPT_TYPE_INTERRUPT,       //
+    .priority = INTERRUPT_PRIORITY_NORMAL,  //
+  };
+  status_ok_or_return(gpio_init_pin(ready_pin, &gpio_settings));
+  status_ok_or_return(gpio_it_register_interrupt(ready_pin, &it_settings, INTERRUPT_EDGE_RISING,
+                                                 prv_interrupt_handler, storage));
+  // Mask the interrupt until channels are enabled by the user.
+  return gpio_it_mask_interrupt(ready_pin, true);
+}
+
+// This function enable/disables channels, and registers callbacks for each channel.
+StatusCode ads1015_configure_channel(Ads1015Storage *storage, Ads1015Channel channel, bool enable,
+                                     Ads1015Callback callback, void *context) {
+  if (storage == NULL || channel >= NUM_ADS1015_CHANNELS) {
+    return status_code(STATUS_CODE_INVALID_ARGS);
+  }
+  status_ok_or_return(gpio_it_mask_interrupt(&storage->ready_pin, true));
+
+  uint8_t channel_enable_bitset = storage->channel_enable_bitset;
+  prv_mark_channel_enabled(channel, enable, &storage->channel_enable_bitset);
+  storage->channel_enable_bitset_pending = storage->channel_enable_bitset;
+
+  if ((channel_enable_bitset == ADS1015_BITSET_EMPTY) && enable) {
+    // Activate the interrupt since the first channel is being enabled.
+    status_ok_or_return(prv_set_channel(storage, channel));
+    status_ok_or_return(gpio_it_mask_interrupt(&storage->ready_pin, false));
+  } else if (!enable) {
+    storage->channel_readings[channel] = ADS1015_DISABLED_CHANNEL_READING;
+  }
+  storage->channel_callback[channel] = callback;
+  storage->callback_context[channel] = context;
+  status_ok_or_return(gpio_it_mask_interrupt(&storage->ready_pin, false));
+  if (storage->channel_enable_bitset == ADS1015_BITSET_EMPTY) {
+    status_ok_or_return(gpio_it_mask_interrupt(&storage->ready_pin, true));
+  }
+  return STATUS_CODE_OK;
+}
+
+// Reads raw 12 bit conversion results which are expressed in two's complement format.
+StatusCode ads1015_read_raw(Ads1015Storage *storage, Ads1015Channel channel, int16_t *reading) {
+  if (channel >= NUM_ADS1015_CHANNELS || storage == NULL || reading == NULL) {
+    return status_code(STATUS_CODE_INVALID_ARGS);
+  }
+  if (!channel_is_enabled(storage, channel)) {
```

Comment:
You might as well combine this with your input check or use `status_msg` to provide an error message.